src/aligned_memory.rs

```diff
@@ -0,0 +1,75 @@
+//! Aligned memory
+
+/// Provides u8 slices at a specified alignment
+#[derive(Clone, Debug, PartialEq)]
+pub struct AlignedMemory {
+    len: usize,
+    align_offset: usize,
+    write_index: usize,
+    mem: Vec<u8>,
+}
+impl AlignedMemory {
+    /// Return a new AlignedMem type
+    pub fn new(len: usize, align: usize) -> Self {
+        let mem = vec![0u8; len + align];
+        let align_offset = mem.as_ptr().align_offset(align);
+        Self {
+            len,
+            align_offset,
+            mem,
+            write_index: align_offset,
+        }
+    }
+    /// Get the length of the data
+    pub fn len(&self) -> usize {
+        self.len
+    }
+    /// Is the memory empty
+    pub fn is_empty(&self) -> bool {
+        self.len == 0
+    }
+    /// Get the current write index
+    pub fn write_index(&self) -> usize {
+        self.write_index
+    }
+    /// Get an aligned slice
+    pub fn as_slice(&self) -> &[u8] {
+        &self.mem[self.align_offset..self.align_offset + self.len]
+    }
+    /// Get an aligned mutable slice
+    pub fn as_slice_mut(&mut self) -> &mut [u8] {
+        &mut self.mem[self.align_offset..self.align_offset + self.len]
+    }
+    /// Fill memory with value starting at the write_index
+    pub fn fill(&mut self, num: usize, value: u8) -> std::io::Result<()> {
+        if self.write_index + num >= self.len {
+            return Err(std::io::Error::new(
+                std::io::ErrorKind::Other,
```

Comment:
How about `std::io::ErrorKind::WriteZero`?

Reply:
Yeah, thanks for pointing that out, threw those in there